mosu-app/src/widgets/auth/SchoolInfoSearchInput.tsx

```diff
@@ -0,0 +1,201 @@
+import { useOverlay } from "@mosu/overlay";
+import { Building2, Loader2, MapPin, Search } from "lucide-react";
+import { useState } from "react";
+
+import { SchoolInfoResponseBody, getSchoolInfo } from "@/entities/profile/services/get-school-info";
+
+import { Input } from "@/shared/components/Input";
+import { Button } from "@/shared/ui/button";
+import { Label } from "@/shared/ui/label";
+
+interface SchoolSearchInputProps {
+    label: string;
+    placeholder?: string;
+    value: string;
+    onChange: (value: string) => void;
+    onSchoolSelect?: (school: SchoolInfoResponseBody) => void;
+    required?: boolean;
+    error?: string;
+}
+
+export const SchoolSearchInput = ({
+    label,
+    placeholder = "학교명을 입력해주세요.",
+    value,
+    onChange,
+    onSchoolSelect,
+    required = false,
+    error,
+}: SchoolSearchInputProps) => {
+    const [searchTerm, setSearchTerm] = useState("");
+    const [schools, setSchools] = useState<SchoolInfoResponseBody[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const overlay = useOverlay({
+        title: "학교 검색",
+        closeOnEnter: false,
+        closeOnEscape: true,
+        closeOnBackdrop: false,
+    });
+
+    const handleSearch = async () => {
+        if (!searchTerm.trim() || searchTerm.length < 2) {
+            console.log("검색어가 너무 짧습니다:", searchTerm);
+            return;
+        }
+
+        setIsLoading(true);
```

Comment:
```suggestion
setIsLoading(true);
setSchools([]); // 결과 초기화
```

색 시작 시 이전 검색 결과가 잠깐이라도 남아 있을 수 있어서 UX 혼란 방지 위해 `setSchools([])` 선행 필요할 것 같아요!

Reply:
아 정말 잠깐이라도 남아있을 수 있겠네요